### PR TITLE
Adds simple demo script

### PR DESCRIPTION
**Pull request by mchurichi**
**Pull Request check list**

- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

**Description of change**
Adds a set of simple demo scripts to demostrate how to setup federation between two SPIRE Server using Galadriel.

There is a Dockerfile and some make targets that are unrelated to the demo but might be useful.

**Which issue this PR fixes**

### CODE REVIEW

**Review by mgbcaio — APPROVED** [review]
Played around a little bit and it looks awesome! Fantastic work, Maxi.

File: demo/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Demo
+
+The following demo script shows how to setup two simple SPIRE deployments and have them federated using Galadriel.
+
+![Demo diagram](diagram.png "Demo diagram")
+
+## Requirements
+
+This demo only requires a `git` client and the `make` command properly installed in the system.
+
```
Comment (Victorblsilveira) [inline demo/README.md:10]: There are any know restrictions related to environments like OS?

**Review by Victorblsilveira — DISMISSED** [review]
Awesome! Just a Quick doubt raised.
LGTM.

File: demo/README.md (hunk 1 above)
Comment (mchurichi) [reply demo/README.md:10]: That's a good question. I only tested on macOS, but as this is using standard bash it should also run on any Linux flavor and WSL, as long they have a relatively new version of bash. You are on Linux, right?

Comment (mgbcaio) [reply demo/README.md:10]: I only tested in macOS as well.

Comment (Victorblsilveira) [reply demo/README.md:10]: Does worth it to specify the environment in which we test it?

**Comment by mchurichi** [comment]
- Added a small note about what environments were used for testing.
- Added the ability to specify the SPIRE version/branch/tag we want to built. The default is the latest (v1.4.4) but it's currently [broken because of a dependency](https://github.com/spiffe/spire/issues/3506). In the meantime the main branch can be used with `SPIRE_VERSION=main ./0-setup.sh`

**Review by Victorblsilveira — DISMISSED** [review]
LGTM

**Review by mgbcaio — APPROVED** [review]
🦖

**Review by jufantozzi — APPROVED** [review]
🥇

**Comment by mchurichi** [comment]
Sorry folks, the automated branch update from github wasn't a signed commit so I had to squash it 🤦 I'll disable that feature to avoid the same issue in the future.

**Review by mgbcaio — APPROVED** [review]

**Review by jufantozzi — APPROVED** [review]

**Comment by mchurichi** [comment]
god 🤦 this is the good one

**Review by mgbcaio — APPROVED** [review]
I got'chu

**Review by jufantozzi — APPROVED** [review]
lul

**Pull request merged**